### PR TITLE
mistral ai initial notebook

### PR DESCRIPTION
**Pull request by chedim**

### CODE REVIEW

**Comment by teetangh** [comment]
Hey Dima, just the notebook should be sufficient. i guess we shoulnd't bulk the repo with config, binary, static files, other build files.

File: mistralai/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,110 @@
+annotated-types==0.7.0
+anyio==4.4.0
```
Comment (teetangh) [inline mistralai/requirements.txt:2]: For the requirements.txt, we need to only keep the requirements that are bare minimum to run the project.

**Comment by teetangh** [comment]
Please add elaborations for each of the cells so that the user can understand what is going on in the cell

File: mistralai/pyvenv.cfg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+home = /Users/dmitrii.chechetkin/micromamba/bin
```
Comment (teetangh) [inline mistralai/pyvenv.cfg:1]: not sure if this file required in repo. if required we shouldn't hard code the credentials with your name

File: mistralai/fts_index.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
+{
+ "name": "vector_test",
+ "type": "fulltext-index",
+ "params": {
+  "mapping": {
+   "types": {
+    "_default._default": {
```
Comment (teetangh) [inline mistralai/fts_index.json:7]: if we follow `_default` scope and `_default` collection, we are going to mix the embedding data from multiple providers.They might not even compatible due to different dimensions like 1536 for cohere and 768 for voyage. 
Suggestion: `shared`.`mistral` , `shared`.`huggingface` , etc.

**Comment by teetangh** [comment]
we definitely shouldn't hardcode here. Rather we can use input and getpass to get the user data.
image

**Comment by teetangh** [comment]
Fix this error
image

**Comment by teetangh** [comment]
Again, declare reusable variables
image
image

**Comment by teetangh** [comment]
Great work on the RAG response. We also need to demonstrate semnatic search and how caching using Couchbase improves performance

**Comment by lokesh-couchbase** [comment]
Hey @chedim, I went through the code. We may require a bit more explanation on how things are working. Best way is to link some docs/tutorials which are already created. e.g For Couchbase Connection, we can share doc as well as python quickstart. For RAG we can point out to Langchain rag demo

**Comment by chedim** [comment]
> Fix this error image

I don't think putting an API key in there would be a good idea?

**Review by lokesh-couchbase — APPROVED** [review]
LGTM

**Pull request merged**